crates/primitives/src/chain/spec.rs

```diff
@@ -2320,9 +2321,49 @@ Post-merge hard forks (timestamp based):
                     Head { number: 0, ..Default::default() },
                     ForkId { hash: ForkHash([0xca, 0xf5, 0x17, 0xed]), next: 3950000 },
                 ),
-                // TODO: complete these, see https://github.com/paradigmxyz/reth/issues/8012
+                // Berlin
                 (
-                    Head { number: 105235063, timestamp: 1710374401, ..Default::default() },
+                    Head { number: 3950000, ..Default::default() },
+                    ForkId { hash: ForkHash([0x52, 0x6a, 0x21, 0x71]), next: 105235063 },
+                ),
+                // London
+                (
+                    Head { number: 105235063, ..Default::default() },
+                    ForkId { hash: ForkHash([0xe3, 0x39, 0x8d, 0x7c]), next: 1704992401 },
+                ),
+                // Arrow Glacier
+                (
+                    Head { number: 105235063, ..Default::default() },
+                    ForkId { hash: ForkHash([0xe3, 0x39, 0x8d, 0x7c]), next: 1704992401 },
+                ),
+                // Grey Glacier
+                (
+                    Head { number: 105235063, ..Default::default() },
+                    ForkId { hash: ForkHash([0xe3, 0x39, 0x8d, 0x7c]), next: 1704992401 },
+                ),
+                // Paris
+                (
+                    Head { number: 105235063, ..Default::default() },
+                    ForkId { hash: ForkHash([0xe3, 0x39, 0x8d, 0x7c]), next: 1704992401 },
+                ),
+                // Bedrock
+                (
+                    Head { number: 105235063, ..Default::default() },
+                    ForkId { hash: ForkHash([0xe3, 0x39, 0x8d, 0x7c]), next: 1704992401 },
+                ),
+                // Canyon
+                (
+                    Head { timestamp: 1704992401, ..Default::default() },
+                    ForkId { hash: ForkHash([0x0, 0x0, 0x0, 0x0]), next: 1708560000 },
+                ),
+                // Delta
+                (
+                    Head { timestamp: 1708560000, ..Default::default() },
+                    ForkId { hash: ForkHash([0x0, 0x0, 0x0, 0x0]), next: 1710374401 },
+                ),
```

Comment:
re: https://github.com/paradigmxyz/reth/issues/8012#issuecomment-2095634418

> Is the forkhash supposed to be 00000000, do you have a source on that? That sounds incorrect

It's not supposed to be all zeros, typically values for this would be obtained from writing a similar test in geth, but since our final (ecotone) forkhash is compliant with geth, what we compute in test failures is probably also correct. However, starting at Canyon we can't use `..Default::default()` because the block needs to be higher than the london / bedrock block. Let's change it back to the previous block number

Reply:
`[0x0, 0x0, 0x0, 0x0]` is just temporary since test failure output is not too useful, to be able to find back to the test case that failed

```
thread 'chain::spec::tests::op_mainnet_forkids' panicked at crates/primitives/src/chain/spec.rs:1643:13:
assertion `left == right` failed: Expected fork ID ForkId { hash: ForkHash("00000000"), next: 1708560000 }, computed fork ID ForkId { hash: ForkHash("caf517ed"), next: 3950000 } at block 0
  left: ForkId { hash: ForkHash("00000000"), next: 1708560000 }
 right: ForkId { hash: ForkHash("caf517ed"), next: 3950000 }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```